lib/src/component_declaration/component_base/component_base_2.dart

```diff
@@ -330,17 +330,45 @@ abstract class UiComponent2<TProps extends UiProps> extends react.Component2
 ///         )(props.children);
 ///       }
 ///     }
-abstract class UiStatefulComponent2<TProps extends UiProps,
-        TState extends UiState> extends UiComponent2<TProps>
-    implements UiStatefulComponent<TProps, TState> {
-  // ----------------------------------------------------------------------
-  // ----------------------------------------------------------------------
-  //   BEGIN Typed state helpers
-  //
+abstract class UiStatefulComponent2<TProps extends UiProps, TState extends UiState>
+    extends UiComponent2<TProps>
+    with UiStatefulMixin2<TProps, TState>
+    implements UiStatefulComponent<TProps, TState> {}
 
-  /// A typed state object corresponding to the current untyped state Map ([unwrappedState]).
+/// A mixin that allows you to add typed state to a [UiComponent2].
+///
+/// Useful for when you're extending from another [UiComponent2] that doesn't extend from [UiStatefulComponent2].
+///
+/// Includes support for strongly-typed [UiState] in-addition-to the
+/// strongly-typed props and utilities for prop and CSS classname forwarding provided by [UiComponent2].
+///
+/// __Initializing state:__
+///
+///     @Component2()
+///     class YourComponent extends UiComponent2<YourProps> with UiStatefulMixin2<YourProps, YourState> {
+///       @override
+///       void init() {
+///         this.state = (newState()
```

Comment:
https://github.com/Workiva/over_react/pull/325

Reply:
👍 We can update this comment once we get all the state stuff figured out

Reply:
Incoming ticket "Finalize "Finalize ComponentV2 / UiComponentV2""